app/views/posts/show.html.haml

```diff
@@ -1,7 +1,7 @@
 - if @pages.respond_to?(:total_pages)
   - content_for(:rel_next_prev_link_tags) { rel_next_prev_link_tags(@pages) }
   - content_for(:canonical_tag) do
-    %link{rel: :canonical, href: url_for(all: true, only_path: false)}
+    %link{rel: :canonical, href: url_for(all: true, only_url: false)}
```

Comment:
http://api.rubyonrails.org/classes/ActionDispatch/Routing/UrlFor.html#method-i-url_for

`only_path: false` is correct for this purpose.